tensorflow_datasets/image/deep_weeds.py

```diff
@@ -98,19 +101,38 @@ def _split_generators(self, dl_manager):
             url=_URL,
             extract_method=tfds.download.ExtractMethod.ZIP))
 
+
+    path_lbl = dl_manager.download_and_extract(
+        tfds.download.Resource(
+            url=_URL_LBL,
+            extract_method=None))
+	
+
+    # there are different label set for train and test
+    # for now we return the full dataset as 'train' set.
     return [
         tfds.core.SplitGenerator(
             name="train",
             gen_kwargs={
                 "data_dir_path": path,
+                "label_dir_path": path_lbl,
             },
         ),
     ]
 
-  def _generate_examples(self, data_dir_path):
+  def _generate_examples(self, data_dir_path, label_dir_path):
     """Generate images and labels for splits."""
-
+    # parse the csv-label data
+    csv = np.loadtxt(label_dir_path,
+            dtype={'names': ('Filename', 'Label', 'Species'), 'formats': ('S21', 'i4', 'S1')},
+            skiprows=1,
+            delimiter=',')
+    
+    label_dict = {}
```

Comment:
The labels that you are adding won't be correct, as the `self.info.features['label'].names` order do not match the ones from the file (another bug from the person which added the dataset).

Instead, you can dynamically assign the labels from the csv file:

In `_info`, only define the number of classes
```python
tfds.features.ClassLabel(num_classes=xx)
```

In `_generate_examples`, update the features labels, after having loaded the csv file:
```python
with tf.io.gfile.GFile(label_path) as f:
  reader = csv.DictReader(f)

  # Extract the mapping int -> str and save the label name string to the feature
  label_id_to_name = {
      row['Label']: row['Species'] for row in reader
  }
  self.info.features['label'].names = [v for k, v in sorted(label_id_to_name.items())]
```

Then you can yield the label string directly, without converting them to int first
```python
filename_to_label = {
    rows['Filename']: row['Species'] for row in reader
}
for file_name in tf.io.gfile.listdir(data_dir_path):
  yield file_name, {
      "image": os.path.join(data_dir_path, file_name), 
      "label": filename_to_label[file_name]
  }
```